python/ray/workflow/step_executor.py

```diff
@@ -183,7 +177,9 @@ def execute_workflow(workflow: "Workflow") -> "WorkflowExecutionResult":
         if step_options.step_type == StepType.WAIT:
             executor = _workflow_wait_executor
         else:
-            executor = _workflow_step_executor
+            # Tell the executor that we are running inplace. This enables
```

Comment:
Should we check the resource requirement for this step? For example, what if the num_cpus are different?

Reply:
It was banned before we could not find a totally valid check: https://github.com/ray-project/ray/pull/19928#discussion_r745385992